lib/index.js

```diff
@@ -129,35 +176,57 @@ export default {
 
         const executable = await this.getExecutablePath(fileDirectory)
 
-        let result
-        try {
-          result = await exec(executable, ['status', '--json'], {
-            cwd: fileDirectory,
-            timeout: 60 * 1000,
-            uniqueKey: 'flow-ide-linter',
-            ignoreExitCode: true,
-          })
-          if (result === null) {
-            return null
-          }
-        } catch (error) {
-          if (error.message.indexOf(INIT_MESSAGE) !== -1 && configFile) {
-            spawnedServers.add(Path.dirname(configFile))
-          }
-          if (error.message.indexOf(INIT_MESSAGE) !== -1 || error.message.indexOf(RECHECKING_MESSAGE) !== -1) {
-            return linter.lint(textEditor)
-          } else if (error.code === 'ENOENT') {
-            throw new Error('Unable to find `flow` executable.')
-          } else {
-            throw error
-          }
+        const version = await this.getVersion(configFile, executable)
+        if (!version) {
+          return null
         }
 
-        return toStatusLinterMessages(result)
+        return semver.gte(version, '0.66.0')
+          ? this.lintV2(textEditor, fileDirectory, executable)
+          : this.lintV1(textEditor, fileDirectory, executable)
       },
     }
     return linter
   },
+  async lintV1(textEditor: TextEditor, fileDirectory: string, executable: string) {
+    let result
+    try {
+      result = await exec(executable, ['status', '--json'], {
+        cwd: fileDirectory,
+        timeout: 60 * 1000,
+        uniqueKey: 'flow-ide-linter',
+        ignoreExitCode: true,
+      })
+      if (result === null) {
```

Comment:
Nit: You can put this outside the try/catch to make the extra block as small as possible